builds/jenkins-build-sls-app/Jenkinsfile

```diff
@@ -0,0 +1,1062 @@
+
+node() {
+
+  def jazzBuildModuleURL = getBuildModuleUrl()
+  loadBuildModules(jazzBuildModuleURL)
+
+  def jazz_prod_api_id = utilModule.getAPIIdForCore(configLoader.AWS.API["PROD"])
+  g_base_url = "https://${jazz_prod_api_id}.execute-api.${configLoader.AWS.REGION}.amazonaws.com/prod"
+
+  echo "Build triggered via branch: " + params.scm_branch + " with params " + params
+  def _event = ""
+
+  def branch = params.scm_branch
+  def repo_name = params.service_name
+  def gitCommitOwner
+  def gitCommitHash
+  def context_map
+  def environment_logical_id
+
+  if (domain && domain != "") {
+    repo_name = params.domain + "_" + params.service_name
+  }
+
+  def domain = params.domain
+
+  stage('Checkout code base') {
+
+    sh 'rm -rf ' + repo_name
+    sh 'mkdir ' + repo_name
+    sh 'pwd'
+
+    def repocloneUrl
+    if (domain && domain == "jazz") {
+      repocloneUrl = scmModule.getCoreRepoCloneUrl(repo_name)
+    } else {
+      auth_token = setCredentials()
+      repocloneUrl = scmModule.getRepoCloneUrl(repo_name)
+    }
+
+    dir(repo_name) {
+      checkout([$class: 'GitSCM', branches: [
+        [name: '*/' + params.scm_branch]
+      ], doGenerateSubmoduleConfigurations: false, extensions: [], submoduleCfg: [], userRemoteConfigs: [
+          [credentialsId: repo_credential_id, url: repocloneUrl]
+        ]])
+    }
+
+    def configObj = dir(repo_name) {
+      LoadConfiguration()
+    }
+
+    if (configObj.service_id) {
+      config = serviceMetadataLoader.loadServiceMetadata(configObj.service_id)
+    } else {
+      error "Service Id is not available."
+    }
+  }
+
+  if (!config) {
+    error "Failed to fetch service metadata from catalog"
+  }
+
+  dir(repo_name) {
+    scmModule.setServiceConfig(config)
+    def envApi = "${g_base_url}/jazz/environments"
+    environmentDeploymentMetadata.initialize(config, configLoader, scmModule, branch, env.BUILD_URL, env.BUILD_ID, envApi, auth_token)
+    gitCommitHash = scmModule.getRepoCommitHash()
+    gitCommitOwner = scmModule.getRepoCommitterInfo(gitCommitHash)
+    context_map = [created_by: config['created_by'], deployed_by: gitCommitOwner]
+    environment_logical_id = '';
+
+    if (branch == 'master') {
+      current_environment = 'prod'
+      environment_logical_id = 'prod';
+    } else {
+      current_environment = 'dev'
+      environment_logical_id = environmentDeploymentMetadata.getEnvironmentLogicalId();
+    }
+
+    if (!environment_logical_id && config['domain'] != 'jazz') {
+      error "The environment has not been created yet and its missing the environment id"
+    }
+  }
+  if (!events) {
+    error "Can't load events module"
+  } //Fail here
+  def eventsApi = "${g_base_url}/jazz/events"
+  events.initialize(configLoader, config, "SERVICE_DEPLOYMENT", branch, environment_logical_id, eventsApi)
+
+  def runtime = config['providerRuntime']
+  def service = config['service']
+  def isScheduleEnabled = isEnabled(config, "eventScheduleRate")
```

Comment:
Config will never have these (`eventScheduleRate`, `event_source_dynamodb` etc.) keys for services of type `sls-app`. Remove these & the code where there is logic using these key/value pairs

Reply:
Removing an unused function that is getLambdaARN

Reply:
Renaming buildLambda(...) method name into buildSlsApp(...) to reflect the real nature of this function

Reply:
Removing isLambdaUpdateRequired variable because it is not used anywhere inside the code

Reply:
Removing isEventSchdld because it is not used anywhere inside the code. Also removing the handleError functions that are not used anymore

Reply:
Resolved with: https://github.com/tmobile/jazz/pull/829/commits/2f79f0802ed7defec88beec897041f49538d7280